hibernate-core/src/main/java/org/hibernate/event/internal/EmptyEventManager.java

```diff
@@ -0,0 +1,232 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.event.internal;
+
+import org.hibernate.cache.spi.Region;
+import org.hibernate.cache.spi.access.CachedDomainDataAccess;
+import org.hibernate.engine.spi.EntityEntry;
+import org.hibernate.engine.spi.SharedSessionContractImplementor;
+import org.hibernate.event.spi.AutoFlushEvent;
+import org.hibernate.event.spi.EventManager;
+import org.hibernate.event.spi.HibernateEvent;
+import org.hibernate.event.spi.FlushEvent;
+import org.hibernate.persister.collection.CollectionPersister;
+import org.hibernate.persister.entity.EntityPersister;
+
+public class EmptyEventManager implements EventManager {
```

Comment:
In order to have the best performance, I think it's best to not have this class and return `null` from `getEventManager()` for the "empty" case. If only a single implementation of `EventManager` is loaded at runtime (or provably exists on the classpath for native), the compiler can inline or at least de-virtualize most/all of the calls. I know that this means we have to do null checks at the use-sites, but unless @Sanne can assure us that having such an `EmptyEventManager` implementation on the classpath is no big deal when wanting to use JFR or OpenTelemetry, I prefer to play this safe.

Reply:
From my memory 2 impls is still safe.  But sure, let's see if @Sanne or any more perf-minded people chime in.

I get what you are saying, but also not a fan of null checks sprinkled all across the code base.

Reply:
In such cases it's hard to be 100% certain w/o a thourough comparison and benchmarking, but I'd be very surprised if the `EmptyEventManager` approach would be a problem: invoking a no-op method is very easy to optimise by the compiler.
So I'd go with the cleaner and more maintainable approach initially; we can always revise things if we're proven wrong: it just seems so unlikely that I'd not spend time in proactively verifying this.

You make a good point about the multiple implementations; thinking about that, it might be slightly better to mark this class as `final` and avoid loading it unless it's being used: currently since it's a singleton, any client code referring to `EmptyEventManager.INSTANCE` will trigger loading of this class. It would be better to avoid the singleton pattern in this case and simply construct an `new EmptyEventManager()` whenever one is really needed.